pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go

```diff
@@ -39,19 +39,52 @@ type ChaosEngineSpec struct {
 	//Monitor Enable Status
 	Monitoring bool `json:"monitoring,omitempty"`
 	//JobCleanUpPolicy decides to retain or delete the jobs
-	JobCleanUpPolicy string `json:"jobCleanUpPolicy,omitempty"`
+	JobCleanUpPolicy CleanUpPolicy `json:"jobCleanUpPolicy,omitempty"`
 	//AuxiliaryAppInfo contains details of dependent applications (infra chaos)
 	AuxiliaryAppInfo string `json:"auxiliaryAppInfo,omitempty"`
 	//EngineStatus is a requirement for validation
-	EngineState string `json:"engineState"`
+	EngineState EngineStateString `json:"engineState"`
```

Comment:
```suggestion
	EngineState EngineState `json:"engineState"`
```
If you are using like this then make sure whenever you have to manipulate with EngineState you cannot use string directly you have to use `EngineState` instead of that

Reply:
Yes, @chandankumar4 . This approach seems better, rather than checking the string directly.